### PR TITLE
Replace new/delete and malloc/free with Poco::Buffer for temp buffers

### PR DESCRIPTION
**Pull request by dashodanger**
Poco::Buffer will deallocate its buffer when going out of scope, so is convenient when processing a file/etc and not having to add "delete[]" or "free" to every branch that may return or exit the function.

Also cleaned up some unneeded/unused functions found when tracking down temp buffer usage.

### CODE REVIEW

File: source/engine/client/r_plane.cpp [hunk 1]
```diff
@@ -606,49 +606,6 @@ void R_DrawPlanes(void)
                 flatheight = (float)tex->getHeight();
                 flatwidth = (float)tex->getWidth();
 
-                // [RH] warp a flat if desired
```
Comment (pbdot) [inline source/engine/client/r_plane.cpp:609]: I've seen this a few times, did odamex comment this out or did we?

File: source/engine/client/sdl/i_system.cpp [hunk 2]
```diff
@@ -510,200 +510,6 @@ void I_SetTitleString(const char *title)
 #include <X11/Xlib.h>
 #endif
 
-//
-// I_GetClipboardText
-//
-// by Denis Lukianov - 20 Mar 2006
-// Cross-platform clipboard functionality
-//
-std::string I_GetClipboardText()
```
Comment (pbdot) [inline source/engine/client/sdl/i_system.cpp:519]: Woowoo, see ya later unneeded code

File: source/engine/common/c_cvars.cpp [hunk 3]
```diff
@@ -635,18 +635,18 @@ BEGIN_COMMAND(set)
     {
         cvar_t *var, *prev;
 
-        var = cvar_t::FindCVar(argv[1], &prev);
```
Comment (pbdot) [inline source/engine/common/c_cvars.cpp:638]: These flips from 0-1 are intended, just seems strange, and not looking to close to understand this morning :)

File: source/engine/common/p_switch.cpp [hunk 4]
```diff
@@ -144,8 +145,6 @@ void P_InitSwitchList(void)
         numswitches   = i / 2;
         switchlist[i] = TextureManager::NOT_FOUND_TEXTURE_HANDLE;
     }
-
-    delete[] alphSwitchList;
```
Comment (pbdot) [inline source/engine/common/p_switch.cpp:148]: Poco::Buffer also means no mismatch on delete vs delete [], which isn't even a compiler error

File: source/engine/common/r_data.cpp [hunk 5]
```diff
@@ -112,24 +112,11 @@ void R_PrecacheLevel(void)
     texturemanager.getTexture(sky2texture);
 
     // Precache sprites
-    uint8_t *hitlist = new uint8_t[numsprites];
```
Comment (pbdot) [inline source/engine/common/r_data.cpp:115]: Hm,  we're removing hitlist here?  Another one where I don't quite understand the logic of what this is doing, though want to raise it

**Review by pbdot — COMMENTED** [review]
Great stuff!  A few comments/questions

File: source/engine/common/c_cvars.cpp (hunk 3 above)
Comment (dashodanger) [reply source/engine/common/c_cvars.cpp:638]: Yeah, the weird indexing was in relation to stuff like this:
`com->argv         = argv - 1; // Hack`

**Pull request merged**